scoop/src/main/java/com/lyft/scoop/Scoop.java

```diff
@@ -30,7 +32,10 @@ private Scoop(String name, Scoop parent, Map<String, Object> services) {
     }
 
     public void destroy() {
-        for (Map.Entry<String, Scoop> entry : this.children.entrySet()) {
+        final Set<Map.Entry<String, Scoop>> entries = this.children.entrySet();
+
+        final Set<Map.Entry<String, Scoop>> entriesCopy = new HashSet<>(entries);
+        for (Map.Entry<String, Scoop> entry : entriesCopy) {
```

Comment:
@buildbreaker This could be simplified
```
for (Scoop value : entriesCopy.values()) {
  value.destroy();
}
```
Or iterate over a list of scoops:
`List<Scoop> valuesCopy = new ArrayList(entries.values())`

It's already merged though so it's ok.